src/main/java/org/gridsuite/sensitivityanalysis/server/service/nonevacuatedenergy/NonEvacuatedEnergyWorkerService.java

```diff
@@ -0,0 +1,936 @@
+                referenceGroupId = entry.getKey();
+                sensitivityCoeff.put(referenceGroupId, 1.);
+                referenceSensitivity = entry.getValue();
+                i++;
+            } else {  // other generator : compute sensitivity coefficient from sensitivity value and reference generator sensitivity value
+                sensitivityCoeff.put(entry.getKey(), entry.getValue() / referenceSensitivity);
+            }
+        }
+
+        // compute the generators cappings needed to set the monitored branch under the limit
+        Map<EnergySource, Double> variationsByEnergySource = new EnumMap<>(EnergySource.class);
+        double injectionVariation = computeInjectionVariationFromGeneratorsSensitivities(network, nonEvacuatedEnergyInputs, sensitivities, sensitivityCoeff, delta, variationsByEnergySource, generatorCappings);
+
+        // update sensitivities by energy source in monitored branch result
+        Map<EnergySource, Double> sensitivitiesByEnergySource = new EnumMap<>(EnergySource.class);
+        sensitivities.forEach((key, value) -> {
+            Generator generator = network.getGenerator(key);
+            if (generator != null) {
+                if (!sensitivitiesByEnergySource.containsKey(generator.getEnergySource())) {
+                    sensitivitiesByEnergySource.put(generator.getEnergySource(), value);
+                } else {
+                    sensitivitiesByEnergySource.put(generator.getEnergySource(), sensitivitiesByEnergySource.get(generator.getEnergySource()) + value);
+                }
+            }
+        });
+        monitoredBranchDetailResult.getSensitivityByEnergySource().putAll(sensitivitiesByEnergySource);
+
+        return injectionVariation;
+    }
+
+    private double computeInjectionVariationFromGeneratorsSensitivities(Network network,
+                                                                        NonEvacuatedEnergyInputs nonEvacuatedEnergyInputs,
+                                                                        Map<String, Double> generatorsSensitivities,
+                                                                        Map<String, Double> sensitivitiesCoefficients,
+                                                                        double delta,
+                                                                        Map<EnergySource, Double> variationsByEnergySource,
+                                                                        Map<String, GeneratorCapping> generatorCappings) {
+        double injectionVariation = 0;
+
+        // compute the coefficient sum
+        double coeffSum = 0;
+        for (Map.Entry<String, Double> entry : generatorsSensitivities.entrySet()) {
+            coeffSum += entry.getValue() * sensitivitiesCoefficients.get(entry.getKey());
+        }
+        //double referenceGeneratorVariation = abs(100 * delta / coeffSum); in CVG ????
+        double referenceGeneratorVariation = abs(delta / coeffSum);
```

Comment:
how do we explain this 100 factor difference with the CVG implementation?! maybe we can change the comment to make it clear

Reply:
coment line removed : to my opinion, it has no sense to multiply here the delta by 100 ...